app/controllers/locationOfGoods/AddressController.scala

```diff
@@ -51,41 +50,66 @@ class AddressController @Inject() (
 
   private type Request = SpecificDataRequestProvider1[Country]#SpecificDataRequest[_]
 
-  private def country(implicit request: Request): Country = request.arg
-
-  private def form(isPostalCodeRequired: Boolean)(implicit request: Request): Form[DynamicAddress] =
-    formProvider("locationOfGoods.address", isPostalCodeRequired)
+  private def form(isPostalCodeRequired: Boolean)(implicit request: MandatoryDataRequest[_]): Form[DynamicAddress] =
+    formProvider("locationOfGoods.address", isPostalCodeRequired)(request.request.messages(messagesApi))
 
   def onPageLoad(departureId: String, mode: Mode): Action[AnyContent] = actions
     .requireData(departureId)
-    .andThen(getMandatoryPage(CountryPage))
     .async {
       implicit request =>
-        countriesService.doesCountryRequireZip(country).map {
-          isPostalCodeRequired =>
-            val preparedForm = request.userAnswers.get(AddressPage) match {
-              case None        => form(isPostalCodeRequired)
-              case Some(value) => form(isPostalCodeRequired).fill(value)
-            }
+        val getCountry: Option[Country] = request.userAnswers
+          .get(CountryPage)
+          .orElse(
+            request.userAnswers.departureData.Consignment.LocationOfGoods.flatMap(
+              _.Address.map(
+                x => Country(CountryCode(x.country), "")
```

Comment:
we don't need the whole country, better than making one with an empty description?